resources/waterheater.rb

```diff
@@ -64,7 +66,9 @@ def self.apply_tankless(model, runner, space, fuel_type, cap, ef,
     new_manager = create_new_schedule_manager(t_set, model, Constants.WaterHeaterTypeTankless)
     new_manager.addToNode(loop.supplyOutletNode)
 
-    new_heater = create_new_heater(Constants.ObjectNameWaterHeater, cap, fuel_type, 1, ef, 0, nil, t_set, space, oncycle_p, offcycle_p, ec_adj, Constants.WaterHeaterTypeTankless, cd, nbeds, model, runner)
+    act_vol = 1
```

Comment:
There are a lot of places where integers are used instead of doubles. Here, height = 48, t_amb = 70, etc. Some were introduced by you and some probably weren't. But can you look through the file and replace them all with doubles? 

Using integers is risky because of integer division where, for example, 5/2 = 2 in ruby. If it's 5.0 or 2.0, but then you get 2.5.